plugins/helm/helm.plugin.zsh

```diff
@@ -14,9 +14,9 @@ command mkdir -p "$ZSH_CACHE_DIR/completions"
 # If the completion file does not exist, generate it and then source it
 # Otherwise, source it and regenerate in the background
 if [[ ! -f "$ZSH_CACHE_DIR/completions/_helm" ]]; then
-  helm completion zsh >| "$ZSH_CACHE_DIR/completions/_helm"
+  helm completion zsh | tee "$ZSH_CACHE_DIR/completions/_helm" >/dev/null
   source "$ZSH_CACHE_DIR/completions/_helm"
 else
   source "$ZSH_CACHE_DIR/completions/_helm"
-  helm completion zsh >| "$ZSH_CACHE_DIR/completions/_helm" &|
+  { helm completion zsh | tee "$ZSH_CACHE_DIR/completions/_helm" >/dev/null } &|
```

Comment:
Are the braces really necessary? I believe all commands of the pipe are run in the background without `{}` no? For instance, this example:
```sh
sleep 10 | tee /dev/null    # waits 10 seconds
sleep 10 | tee /dev/null &  # immediately finishes
sleep 10 | tee /dev/null &| # immediately finishes without output
```

Reply:
BTW documenting this for future eyes, [the docs are a bit confusing here](https://zsh.sourceforge.io/Doc/Release/Shell-Grammar.html#Simple-Commands-_0026-Pipelines):

> If a sublist is terminated by a ‘&’, ‘&|’, or ‘&!’, the shell executes the last pipeline in it in the background, and does not wait for it to finish (note the difference from other shells which execute the whole sublist in the background)

It is not referring to a series of commands piped together, but chained like so:
```sh
cp a b && mv b c && rm a &
```
[See this StackExchange answer](https://unix.stackexchange.com/a/88235/487474) for more.